extensions/guacamole-auth-duo/src/main/java/org/apache/guacamole/auth/duo/UserVerificationService.java

```diff
@@ -71,10 +74,60 @@ public void verifyAuthenticatedUser(AuthenticatedUser authenticatedUser)
         // Pull the original HTTP request used to authenticate
         Credentials credentials = authenticatedUser.getCredentials();
         HttpServletRequest request = credentials.getRequest();
+        IPAddress clientAddr = new IPAddressString(request.getRemoteAddr()).getAddress();
 
         // Ignore anonymous users
         if (authenticatedUser.getIdentifier().equals(AuthenticatedUser.ANONYMOUS_IDENTIFIER))
             return;
+        
+        // We enforce by default
+        boolean enforceHost = true;
```

Comment:
Isn't this almost exactly the same code as in `UserVerificationService.java` below? Can you create a shared function?

Reply:
This would also be a nice place to explicitly document the behavior when an address is on both the bypass _and_ enforce lists.

Reply:
Sure, will try to address both concerns.

Reply:
> Isn't this almost exactly the same code as in `UserVerificationService.java` below? Can you create a shared function?

When you say "below", do you mean the code common to checking both enforce and bypass lists within this class, or were you talking about the code across the classes (shared between Duo and TOTP)? I can certainly create some classes within `guacamole-ext` that could be shared among these - and, honestly, probably other - use-cases.

Reply:
I was mostly talking about [this code block](https://github.com/apache/guacamole-client/pull/911/files#diff-5ce38f439ffddba700c3d8ca58d67bd7d1a2e377f6ac852783b1109df0107d8cR84-R130) from lines 84 to 130, and the similar code block in the TOTP `UserVerificationService.java` (sorry, that was unclear...)

If there's other similar code that you think should be shared, I'm definitely open to that as well.

Reply:
Okay - I took some of the common code and moved it into a utility class in `guacamole-ext`. I was trying to balance collecting common code against putting too much extension-specific code into the `guacamole-ext` code, but if you think there's more that should get consolidated into it, I'm happy to take another go at it.